Feature/messages ui with job system

maxwellpark opened the pull request:
Integrating the messages and jobs systems. 

If a message contains a job offer, a button will appear in the message detail view. Click that button to accept the job. The job will then show up in the jobs panel's available jobs container. Drag it into the schedule container to schedule the job.

maxwellpark committed:
Migrating messages scripts from the old repo

maxwellpark committed:
Message scroll views and message item prefab

maxwellpark committed:
Added message prefabs and fixed message detail view text display

maxwellpark committed:
Button for accepting jobs from the message detail view

maxwellpark committed:
Separate jobs panel for viewing availale jobs and scheduling once they've been accepted

maxwellpark committed:
Event action for accepting jobs. Removed subject from message buttons that appear in the message list

maxwellpark committed:
ScriptableObject for storing schedule (calendar) data pertaining to jobs

maxwellpark committed:
DnD logic for individual schedule slots. Prefab made with text indicating day of month

maxwellpark committed:
Event for scheduling jobs.

maxwellpark committed:
Prevented double bookings.

ambid17 commented on Assets/Scripts/CharacterCustomisation/ColourSliders.cs:
```diff
@@ -21,12 +21,12 @@ public class ColourSliders : MonoBehaviour
-        InitialiseSliderContainer();
-        InitialiseImageObject();
+        //InitialiseSliderContainer();
+        //InitialiseImageObject();
```
bad

ambid17 commented on Assets/Scripts/JobSystem/DragNDrop.cs:
```diff
@@ -49,10 +53,16 @@ public void OnEndDrag(PointerEventData eventData)
+            // If the job was previously scheduled, update the schedule accordingly 
+            if (job.isScheduled)
+            {
+                onUnscheduleJob?.Invoke(job);
```
the name `onUnscheduleJob` seems unintuitive here

maxwellpark replied:
Well this logic fires if the player drags the job out of the schedule. In this scenario, it was already scheduled (already inside the schedule container), then it becomes unscheduled (goes back to the available jobs container). I think it's okay personally. But if you have a better name, I'm all ears. 

The comment is a tad misleading though. Will change that.

ambid17 commented on Assets/Scripts/JobSystem/JobsUI.cs:
```diff
@@ -61,8 +64,8 @@ private GameObject InitialiseScheduleContainer()
-        scheduleObject.AddComponent<CanvasGroup>();
-        scheduleObject.AddComponent<Schedule>(); 
+        scheduleObject.AddComponent<CanvasGroup>().blocksRaycasts = true; 
+        //scheduleObject.AddComponent<ScheduleUI>(); 
```
bad

maxwellpark replied:
Just the commented out line right?

ambid17 commented on Assets/Scripts/JobSystem/JobsUI.cs:
```diff
@@ -61,8 +64,8 @@ private GameObject InitialiseScheduleContainer()
 
-        scheduleObject.AddComponent<CanvasGroup>();
-        scheduleObject.AddComponent<Schedule>(); 
+        scheduleObject.AddComponent<CanvasGroup>().blocksRaycasts = true; 
```
this is a hack if ive ever seen one 😂

maxwellpark replied:
lol it's not exactly ideal, but will it suffice for now?

ambid17 replied:
yeah its fine, it was just funny

ambid17 commented on Assets/Scripts/JobSystem/JobsUI.cs:
```diff
@@ -6,21 +6,20 @@
     private GameObject scheduleContainer;
 
     private Sprite containerBackgroundSprite;
     private Sprite scheduleSlotSprite; 
```
unused

maxwellpark replied:
It is used to initialise the schedule slots: `for (int i = 1; i <= schedule.numberOfDays; i++)` 

But nowhere else it seems. I could move numberOfDays to a static member of a constants class or have it as a member of JobsUI. At the moment, it is stored in the schedule scriptableobject. 

What should I do with it?

ambid17 replied:
oh, i did a `find all references` and it didn't find any usages, must be a VS bug

ambid17 commented on Assets/Scripts/UI/MainTabbedUIManager.cs:
```diff
@@ -7,19 +7,21 @@ public class MainTabbedUIManager : MonoBehaviour
     public enum Tab
     {
-        missions, messages, analytics, crew, upgrades
+        missions, messages, jobs, analytics, crew, upgrades
```
typically enum values are capitalized

maxwellpark committed:
Addressing changes. Adjusting Canvas prefab

maxwellpark committed:
Addressing changes and fixing scrollview

maxwellpark committed:
Cleaning up filesystem

maxwellpark committed:
Make buttons uninteractable if already accepted

maxwellpark committed:
Attempting to only rebuild the empty schedule slots

For now, the script is rebuilding them all, and thus not showing the previously scheduled jobs in the correct container.

maxwellpark committed:
Containers no longer generated at run-time

Solves the problems experienced across powering off the game.

maxwellpark committed:
Slot dimensions

maxwellpark committed:
Revert "Slot dimensions"

This reverts commit ea5a561c27c37f9a8b024c6bc17bbc2c3f4967bb.

maxwellpark committed:
Revert "Containers no longer generated at run-time"

This reverts commit f18140dce7d6df9b0b81a79e9e3fb1682204ee24.

maxwellpark committed:
Saving reference to slot transforms associated with previously scheduled jobs

maxwellpark committed:
Schedule as static dictionary

maxwellpark committed:
Revert "Schedule as static dictionary"

This reverts commit 4f925b928c9c14870b75f9500a305d2cad681956.

maxwellpark committed:
Fixed schedule text VLG issue. Increased mouse wheel scrolling sensitivity.
